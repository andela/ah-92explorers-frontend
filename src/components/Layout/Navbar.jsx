/* eslint-disable global-require */
import React from 'react';
import { Link } from 'react-router-dom';
import Dropdown from './Dropdown.jsx';

const Navbar = ({ token, username, avatar }) => {
  if (!token || token === undefined) {
    return (
      <nav className="feedNavbar" style={{ backgroundColor: '#fffff' }}>
        <div className="firstNavbar">
          <a href="/">
            <img src={require('../../assets/icons/logo.png')} alt="" className="logoNavbar" />
          </a>
          <div className="notifications-drop">
            <ul>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
              <li className="title">
            hillary liked your article
                <span>unread</span>
              </li>
            </ul>
          </div>
          <button className="signupBtnNav disappear dBtnOne" type="button"><Link to="/signup">Sign Up</Link></button>
          <button className="signinBtnNav disappear" type="button"><Link to="/login">Sign In</Link></button>

        </div>
        <div className="secondNavbar">
          <div className="listBtns disappearBtn">
            <button className="signupBtnNav" type="button"><Link to="/signup">Sign Up</Link></button>
            <button className="signinBtnNav" type="button"><Link to="/login">Sign In</Link></button>
          </div>
        </div>
      </nav>
    );
  }
  return (
    <nav className="feedNavbar feedLoggedIn" style={{ backgroundColor: '#fffff' }}>
      <div className="firstNavbar">
        <img src={require('../../assets/icons/logo.png')} alt="" className="logoNavbar" />
        <div className="notifications-drop">
          <ul>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
            <li className="title">
            hillary liked your article
              <span>unread</span>
            </li>
          </ul>
        </div>
        <div className="userStatus">
          <img src={avatar} alt="" className="avatarFeed" />
          <span id="userStatusName">{username}</span>
          <Dropdown />
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
