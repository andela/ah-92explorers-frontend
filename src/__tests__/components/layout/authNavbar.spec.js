import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../../components/Layout/Navbar.jsx';


describe('<Navbar />', () => {
  const props = {
    token: 'something', 
    username: 'username',
    avatar: 'avatar'
  };

  const secondProps = {
    token: undefined, 
    username: 'username',
    avatar: 'avatar'
  };
  it('should render Navbar component', () => {
    let wrapper = shallow(
        <Navbar {...props} />,
      );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Navbar component', () => {
    let wrapper = shallow(
        <Navbar {...secondProps} />,
      );
      expect(wrapper.find('.feedNavbar').length).toBe(1);
  });
});
