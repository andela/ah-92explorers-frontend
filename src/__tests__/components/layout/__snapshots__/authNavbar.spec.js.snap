// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthNavbar /> should render AuthNavbar component 1`] = `
<nav>
  <div
    className="firstNavbar"
  >
    <button
      className="createArt"
      type="button"
    >
      <Link
        to="./createArticle.html"
      >
        <img
          alt=""
          className="articleBtn"
          src="test-file-sdub"
        />
        Write Article
      </Link>
    </button>
    <img
      alt=""
      className="logo"
      src="test-file-sdub"
    />
    <input
      className="search-bar"
      name="search"
      placeholder="Title, author, tag, keyword ..."
      type="text"
    />
    <div>
      <img
        alt=""
        className="navIcons notifications disappear"
        src="test-file-sdub"
      />
      <span
        className="notif-count"
      >
        6
      </span>
    </div>
    <div
      className="notifications-drop"
    >
      <ul>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
        <li
          className="title"
        >
          hillary liked your article
          <span>
            unread
          </span>
        </li>
      </ul>
    </div>
    <img
      alt=""
      className="manIcon"
      src="test-file-sdub"
    />
    <Connect(Dropdown) />
  </div>
</nav>
`;
